eng/scripts/Validate-TypeSpec.ps1

```diff
@@ -5,7 +5,9 @@ param (
   [Parameter(Position = 1, Mandatory = $false)]
   [string]$TargetBranch,
   [Parameter(Position = 2, Mandatory = $false)]
-  [string]$SourceBranch
+  [string]$SourceBranch,
+  [Parameter(Position = 3, Mandatory = $false)]
+  [switch]$GitClean
```

Comment:
Setting Git Clean as false as default because I don't want devs to accidentally run the script and clearing all their efforts